resources/2024-06-support-secretstores-env.md

```diff
@@ -0,0 +1,417 @@
+- Update functional tests to cover the new functionality.
+
+## Security
+
+<!--
+Describe any changes to the existing security model of Radius or security 
+challenges of the features. For each challenge describe the security threat 
+and its mitigation with this design. 
+
+Examples include:
+- Authentication 
+- Storing secrets and credentials
+- Using cryptography
+
+If this feature has no new challenges or changes to the security model
+then describe how the feature will use existing security features of Radius.
+-->
+The handling of secrets will remain within Kubernetes and Radius is only providing a way to reference these secrets in the environment variables of a container. This is an improvement over the current method of mounting secrets as volumes as it allows for more flexibility and security. Also, the secrets are stored in the Kubernetes secret store and are never exposed to the user.
+
+## Compatibility (optional)
+
+<!--
+Describe potential compatibility issues with other components, such as
+incompatibility with older CLIs, and include any breaking changes to
+behaviors or APIs.
+-->
+These will be breaking changes to the schema. Users will need to update the environment variables in their bicep files to use the new secret reference type.
+
+## Monitoring and Logging
+
+<!--
+Include the list of instrumentation such as metric, log, and trace to 
+diagnose this new feature. It also describes how to troubleshoot this feature
+with the instrumentation. 
+-->
+No additional monitoring or logging is required for this feature.
+
+## Development plan
+
+<!--
+Describe how you will deliver your features. This includes aligning work items
+to features, scenarios, or requirements, defining what deliverable will be
+checked in at each point in the product and estimating the cost of each work
+item. Don’t forget to include the Unit Test and functional test in your
+estimates.
+-->
+Work completed in a pair programming session with a second developer. The work will be broken down into the following tasks:
+- Update the versioned datamodel to include the new secret reference type
+- Update the conversion functions to handle the new secret reference type
+- Update the containers typespec to include the new secret reference type
+- Update the common typespec to include the new secret reference type
+- Update the functional tests to cover the new functionality
+- Update the documentation to include the new functionality
+
+## Open Questions
+
+<!--
+Describe (Q&A format) the important unknowns or things you're not sure about. 
+Use the discussion to answer these with experts after people digest the 
+overall design.
+-->
+
+## Alternatives considered
+
+<!--
+Describe the alternative designs that were considered or should be considered.
+Give a justification for why alternative approaches should be rejected if
+possible. 
+-->
+
+## Design Review Notes
+
+<!--
```

Comment:
Please update this section to capture the conclusion from our discussion last week.